lib/src/base-styles/_nx-colors.scss

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+:root {
+  --nx-color-border-default: #{$nx-indigo-100};
+  --nx-color-border-subsection: #{$nx-grey-200};
+  --nx-color-border-form-element: #{$nx-grey-1100};
+  --nx-color-border-form-element-pristine: #{$nx-indigo-100};
+  --nx-color-border-disabled: #{$nx-grey-500};
+  --nx-color-border-focus: #{$nx-teal-300};
+
+  --nx-color-box-shadow-focus: #{transparentize($nx-teal-300, 0.7)};
+
+  --nx-color-background-component: #{$nx-white};
+  --nx-color-background-site: #{$nx-grey-50};
```

Comment:
I think that these should be `nx-color-comonent-background` and `nx-color-site-background`

Reply:
> I think that these should be nx-color-comonent-background and nx-color-site-background

That would be inconsistent with the naming scheme we agreed on

Reply:
> > I think that these should be nx-color-comonent-background and nx-color-site-background
> 
> That would be inconsistent with the naming scheme we agreed on

I disagree. Looking at the other variables the naming is `nx-color-uniqueThing-specificPartOfThing`. The background is not the unique thing, the site or component is.

For example: `--nx-color-border-default`, `--nx-color-link-hover`or `--nx-color-threat-moderate`.

Looking at it another way if the site had a border we wanted to encapsulate in a variable we'd want:
`--nx-color-site-background`
`--nx-color-site-border`

Reply:
> I disagree. Looking at the other variables the naming is nx-color-uniqueThing-specificPartOfThing. The background is not the unique thing, the site or component is.

Actually it's `nx-color-thingBeingColored-state`.  Looking now I see that `thingBeingColored` is sometimes a property (e.g. background), sometimes a "thing" (e.g. link), and sometimes both (`border-form-element`, `background-site`.  I guess really, the full structure that I arrived at (without realizing it) is
```
nx-color-[property-][thing-][state]
```
where _at least one_ of `property` and `thing` are required.

> Looking at it another way if the site had a border we wanted to encapsulate in a variable we'd want:
> `--nx-color-site-background`
> `--nx-color-site-border`
Consistent with what we've got for other variables, I'd want to keep those as `--nx-color-background-site` and `--nx-color-border-site`.

Reply:
> Consistent with what we've got for other variables

I suspect that most of the places where you see this consistency I think that those variables should be renamed.

I think that the number of variables that should be in the form of `nx-color-[property-][state]` is low and should only be considered for things like `nx-border-default` which are intended to be used in many places and are therefore very generic. I think that every other border variable in that list should be in the form of:

`--nx-color-subsection-border`
`--nx-color-form-element-border`
`--nx-color-form-element-border-pristine`

This matches the `text`, `link`, `threat` & `validation` variables below.

Reply:
> I think that the number of variables that should be in the form of nx-color-[property-][state] is low and should only be considered for things like nx-border-default which are intended to be used in many places and are therefore very generic

There's some sort of disconnect in the way we're reading things.  By my count, 21 of the 28 color variables have states - not a small amount.

I feel like what you may be advocating for is reordering the `[property-][thing-]` in the naming scheme to `[thing-][property-]`, and then grouping by `thing` instead of by `property`, and making the `thing` of `interactive` explicit (it currently is implicit).  Doing that results in something like this:

```
:root {
  --nx-color-border: #{$nx-indigo-100};

  --nx-color-subsection-border: #{$nx-grey-200};

  --nx-color-component-background: #{$nx-white};

  --nx-color-site-background: #{$nx-grey-50};

  --nx-color-form-element-border: #{$nx-grey-1100};
  --nx-color-form-element-border-pristine: #{$nx-indigo-100};

  --nx-color-interactive-background-selected: #{$nx-blue-200};
  --nx-color-interactive-background-active: #{$nx-blue-200};
  --nx-color-interactive-background-hover: #{$nx-grey-100};
  --nx-color-interactive-background-disabled: #{$nx-grey-200};
  --nx-color-interactive-border-disabled: #{$nx-grey-500};
  --nx-color-interactive-border-focus: #{$nx-teal-300};
  --nx-color-interactive-box-shadow-focus: #{transparentize($nx-teal-300, 0.7)};

  --nx-color-text-dark: #{$nx-grey-1100};
  --nx-color-text: #{$nx-grey-900};
  --nx-color-text-disabled: #{$nx-grey-500};

  --nx-color-link: #{$nx-blue-600};
  --nx-color-link-active: #{$nx-blue-400};
  --nx-color-link-hover: #{$nx-blue-800};
  --nx-color-link-disabled: #{$nx-grey-500};

  --nx-color-threat-unspecified: #{$nx-indigo-100};
  --nx-color-threat-none: #{$nx-blue-300};
  --nx-color-threat-low: #{$nx-blue-600};
  --nx-color-threat-moderate: #{$nx-yellow-700};
  --nx-color-threat-severe: #{$nx-orange-700};
  --nx-color-threat-critical: #{$nx-red-800};

  --nx-color-validation-valid: #{$nx-green-700};
  --nx-color-validation-invalid: #{$nx-red-800};

}

```

This results in more variables that aren't grouped at all, and makes the `interactive` variables longer, but otherwise I'm be ok with it.

Reply:
> > I think that the number of variables that should be in the form of nx-color-[property-][state] is low and should only be considered for things like nx-border-default which are intended to be used in many places and are therefore very generic
> 
> There's some sort of disconnect in the way we're reading things. By my count, 21 of the 28 color variables have states - not a small amount.
> 
> I feel like what you may be advocating for is reordering the `[property-][thing-]` in the naming scheme to `[thing-][property-]`, and then grouping by `thing` instead of by `property`, and making the `thing` of `interactive` explicit (it currently is implicit). Doing that results in something like this:
> 
> This results in more variables that aren't grouped at all, and makes the `interactive` variables longer, but otherwise I'm ok with it.

Yes, this is exactly what I was suggesting. I agree about the `interactive` variables and I'm open to changing the name, but aside from the length `interactive` works.

Reply:
> Yes, this is exactly what I was suggesting. I agree about the interactive variables and I'm open to changing the name, but aside from the length interactive works.

7a3d0824 and 286afe5e